parlai/core/metrics.py

```diff
@@ -227,15 +237,33 @@ class Metrics(object):
     def __init__(self, opt):
         self.metrics = {}
         self.metrics['cnt'] = 0
-        self.metrics_list = ['mean_rank', 'loss', 'correct', 'f1', 'ppl']
-        if nltkbleu is not None:
-            # only compute bleu if we can
-            self.metrics_list.append('bleu')
-        if rouge is not None:
-            # only compute rouge if we can
-            self.metrics_list.append('rouge-1')
-            self.metrics_list.append('rouge-2')
-            self.metrics_list.append('rouge-L')
+        self.metrics_list = set()
+        optional_metrics_list = []
+        self.compute_rouge_n = 0
+        self.compute_rouge_l = False
+        metrcis_arg = opt.get('metrics', 'default')
+        if metrcis_arg == 'default':
+            optional_metrics_list = __DEFAULT_METRICS__
+        elif metrcis_arg == 'all':
+            optional_metrics_list = __ALL__METRICS__
+        else:
+            optional_metrics_list = set(metrcis_arg.split(','))
+            optional_metrics_list.add('correct')
+        for each_m in optional_metrics_list:
+            if each_m.startswith('rouge') and rouge is not None:
+                self.metrics_list.add(each_m)
+                if each_m == 'rouge-L':
```

Comment:
Hmm, it might be cleaner to just have `rouge`, which computes all the rouge metrics, or doesn't.

Reply:
But we may want some control over which rouge we want to compute? 
1. Rouge-L is time consuming since it involves a LCS computation.
2. You mentioned before we want to make this n a parameter?